Add content to tutorials docs page.

alexfogg opened the pull request:
Follow up to CE RKO breakout - adding docs content to a new user-focused tutorials / getting started page as a landing spot for async learning & enablement.

## Test plan

Docs change only, tested locally.

amenne commented on doc/tutorials/index.md:
```diff
@@ -1,93 +1,73 @@
+| [Using the Sourcegraph CLI](https://docs.sourcegraph.com/cli/quickstart) | How-To-Guide | `src` is a command line interface to Sourcegraph that allows you to search code from your terminal, create and apply batch changes, and manage and administrate repositories, users, and more. |
+| [Saving Searches](https://docs.sourcegraph.com/code_search/how-to/saved_searches) | How-To-Guide | Saved searches let you save and describe search queries so you can easily find and use them again later. You can create a saved search for anything, including diffs and commits across all branches of your repositories. |
+| [Search Contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts) | How-To-Guide | Search contexts help you search the code you care about on Sourcegraph. A search context represents a set of repositories at specific revisions on a Sourcegraph instance that will be targeted by search queries by default.
+ |
```
I think because this is on its own line it's just showing up as text

amenne reviewed (approved):
One tiny thing but this is fantastic and a great starting point to build on! Merge it!

alexfogg commented:
Ooh @amenne one last thing - we need to put a link in the sidebar somewhere. What do you think? Idea in screenshot below.

<img width="1172" alt="Screen Shot 2023-02-16 at 9 47 15 AM" src="https://user-images.githubusercontent.com/2609466/219397918-ccf05f70-62d2-469c-ae4e-36616096e16c.png">

alexfogg commented:
@amenne pushed up a few things

- Added sidebar link "Tutorials"
- Changed "Getting Started" to "Home"
- Linked to 101 and Tours from within the tutorials page
- Misc cleanup (relative link paths, typos)

Just to get this shipped faster I decided to not rename tours right now since it's linked from a couple spots - just wanna get this out the door.
